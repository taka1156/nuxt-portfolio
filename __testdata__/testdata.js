const MAX_NAV_DATA = 4;
const MAX_PROFILE_DATA = 4;
const MAX_CARD_DATA = 6;
const dummyImgUrl = 'http://placehold.jp/150x150.png';
const GITHUB_URL = 'https://github.com/taka1156/audio-player';
const WEBSITE_URL = 'https://takatechblog.netlify.app/';

// factory
const dummyFactory = (num, fn) => {
  return [...new Array(num)].map((_, i) => fn(i + 1));
};

// template
const dummyNavTemplate = () => ({
  name: 'ダミーナビゲーション',
  to: '/test1',
  img: dummyImgUrl,
});

const dummyCardTemplate = () => ({
  title: 'ダミーカード',
  img: { url: dummyImgUrl },
  content2:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
});

const dummyCardLinkTemplate = i => ({
  title: 'ダミーカード',
  img: { url: dummyImgUrl },
  content2:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  link: i % 2 === 0 ? GITHUB_URL : WEBSITE_URL,
});

const dummyProfileTemplate = () => {
  return 'ダミープロフ';
};

const dummyImg = {
  imgUrl: dummyImgUrl,
  imgAlt: 'ダミーの画像',
};

const dummyRoutes = dummyFactory(MAX_NAV_DATA, () => dummyNavTemplate());

const dummyCards = dummyFactory(MAX_CARD_DATA, () => dummyCardTemplate());

const dummyCardsLink = dummyFactory(MAX_CARD_DATA, i => dummyCardLinkTemplate(i));

const dummyProfiles = dummyFactory(MAX_PROFILE_DATA, () => dummyProfileTemplate());

const dummyLogo = 'ダミーロゴ';

export {
  dummyImg,
  dummyLogo,
  dummyRoutes,
  dummyCards,
  dummyCardsLink,
  dummyProfiles,
};
