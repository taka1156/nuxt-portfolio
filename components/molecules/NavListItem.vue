<template>
  <div>
    <li class="nav__item" @click="$emit('change-state')">
      <nuxt-link :to="navItem.to" class="nav__link">
        <nav-img :nav-img="navItem.img" />
        <nav-text :nav-text="navItem.name" />
      </nuxt-link>
    </li>
  </div>
</template>

<script>
import NavImg from '../atoms/NavImg';
import NavText from '../atoms/NavText';

export default {
  name: 'NavListItem',
  components: {
    'nav-img': NavImg,
    'nav-text': NavText,
  },
  props: {
    navItem: {
      type: Object,
      default: () => {},
      required: true,
    },
  },
};
</script>

<style scoped>
.nav__item {
  display: block;
  font-size: 50px;
  text-align: left;
  color: white;
}

.nav__link {
  display: flex;
  width: 100%;
  text-decoration: none;
}
</style>
