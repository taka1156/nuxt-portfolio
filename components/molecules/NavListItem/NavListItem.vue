<template>
  <div class="nav-item" @click="changeState">
    <nuxt-link :to="navItem.to" class="nav-item__link">
      <base-img
        :size="`lg`"
        :img-url="navItem.img"
        :img-alt="`${navItem.name}の画像`"
      />
      <base-text :use-type="`nav`">{{ navItem.name }}</base-text>
    </nuxt-link>
  </div>
</template>

<script>
import BaseImg from '../../atoms/BaseImg/BaseImg';
import BaseText from '../../atoms/BaseText/BaseText';

export default {
  name: 'NavListItem',
  components: {
    'base-img': BaseImg,
    'base-text': BaseText,
  },
  props: {
    /**
     * ナビゲーションの１項目分
     */
    navItem: {
      type: Object,
      default: () => {},
      required: true,
    },
  },
  methods: {
    changeState() {
      /**
       * ナビゲーション開閉の状態を変化させる
       * (NavListにイベント伝搬)
       * @event changeState
       */
      this.$emit('change-state');
    },
  },
};
</script>

<style scoped>
::v-deep .base-text--extend {
  display: inline;
  color: white;
}
.nav-item {
  display: block;
  font-size: 50px;
  text-align: left;
}

.nav-item__link {
  display: flex;
  width: 100%;
  text-decoration: none;
}
</style>
