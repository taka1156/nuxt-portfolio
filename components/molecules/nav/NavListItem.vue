<template>
  <div>
    <li class="nav-list-item" @click="$emit('change-state')">
      <nuxt-link :to="navItem.to" class="nav-list-item__link">
        <nav-img :nav-img="navItem.img" :img-alt="`${navItem.name}の画像`" />
        <base-text class="base-text--nav">{{ navItem.name }}</base-text>
      </nuxt-link>
    </li>
  </div>
</template>

<script>
import NavImg from '../../atoms/nav/NavImg';
import BaseText from '../../atoms/base/BaseText';

export default {
  name: 'NavListItem',
  components: {
    'nav-img': NavImg,
    'base-text': BaseText,
  },
  props: {
    navItem: {
      type: Object,
      default: () => {},
      required: true,
    },
  },
};
</script>

<style scoped>
.nav-list-item {
  display: block;
  font-size: 50px;
  text-align: left;
  color: white;
}

.nav-list-item__link {
  display: flex;
  width: 100%;
  text-decoration: none;
}
</style>
