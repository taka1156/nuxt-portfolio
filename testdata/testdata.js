const dummyLogo = 'ダミー';

const dummyRoutes = [
  {
    name: 'nav-item',
    to: '/test1',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test2',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test3',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test4',
    img: 'http://placehold.jp/150x150.png',
  },
];

const dummyProfile = [
  'TypeScriptの学習',
  'Laravel+Vueでサービスを作る',
  'AWSなどインフラ周りの学習',
  'CSS/SASSの学習',
];

const dummyCards = [
  {
    title: 'PHP',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '1~4ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'Laravel',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '2~3ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'JavaScript(ES6)',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '６ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'Vue.js',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '8ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
];

const dummyCardsLink = [
  {
    title: 'ブログサイト',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Nuxt.js(Vue.js), axios, microCMS, BootStrap4, vue-infinite-loading',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://takatechblog.netlify.app/',
  },
  {
    title: 'マークダウンエディター',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Vue.js, Electron, BootStrap4',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/markdown-editor',
  },
  {
    title: 'LineBot',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Express.js, axios, @line/bot-sdk , Vercel(Zeit Now)',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/line-bot',
  },
  {
    title: 'オーディオプレイヤー',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Vue.js, Electron, JSMediaTags',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/audio-player',
  },
];

export { dummyLogo, dummyRoutes, dummyProfile, dummyCards, dummyCardsLink };
