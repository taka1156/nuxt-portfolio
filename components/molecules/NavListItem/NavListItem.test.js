import { mount, RouterLinkStub } from '@vue/test-utils';
import NavListItem from './NavListItem.vue';
import { dummyRoutes } from '@/__testdata__/testdata.js';

describe('NavListItem', () => {
  const navListItem = mount(NavListItem, {
    stubs: {
      NuxtLink: RouterLinkStub
    },
    propsData: {
      navItem: dummyRoutes[0]
    }
  });

  it('NavList初期値: navItem', () => {
    // NavItme
    expect(navListItem.vm.$options.props.navItem.required).toBe(true);
    expect(navListItem.vm.navItem).toBe(dummyRoutes[0]);
  });

  it('値がDOMに反映されているか', () => {
    const aTag = navListItem.findComponent(RouterLinkStub);
    const imgTag = navListItem.find('img');

    expect(aTag.props().to).toBe(dummyRoutes[0].to);
    expect(aTag.text()).toBe(dummyRoutes[0].name);
    expect(imgTag.attributes().src).toBe(dummyRoutes[0].img);
    // スナップショット
    expect(navListItem.html()).toMatchSnapshot();
  });
});
