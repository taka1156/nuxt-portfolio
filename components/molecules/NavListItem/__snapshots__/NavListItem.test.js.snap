// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavListItem 値がDOMに反映されているか 1`] = `
"<div>
  <div class=\\"nav-item\\"><a class=\\"base-link--extend\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミー1の画像\\" class=\\"base-img--extend base-img--lg\\">
      <p class=\\"base-text--extend\\">ダミー1</p>
    </a></div>
</div>"
`;
